### PR TITLE
validation for verify status

### PR DESCRIPTION
**Pull request by shreya-mishra**
- [x] created new route called /get-qrcode-device-info?user_id = {user_id}
- [x] functions related to that which is filtering according to the authorization status
- [x] Test

### CODE REVIEW

**Review by DashDeipayan — CHANGES_REQUESTED** [review]

File: controllers/auth.js [hunk 1]
```diff
@@ -166,11 +166,29 @@ const fetchUserDeviceInfo = async (req, res) => {
   }
 };
 
+const fetchDeviceDetails = async (req, res) => {
+  try {
+    const userId = req.query.user_id;
```
[the remark this answers is not in the document]
Comment (shreya-mishra) [reply controllers/auth.js]: resolved it in last commit

File: controllers/auth.js [hunk 2]
```diff
@@ -166,11 +166,29 @@ const fetchUserDeviceInfo = async (req, res) => {
...
+    const userDeviceInfoData = await QrCodeAuthModel.retrieveUserDeviceInfo({ userId });
+    if (!userDeviceInfoData.userExists) {
+      return res.boom.notFound(`User with id ${userId} does not exist.`);
+    }
+    return res.json({
+      message: "Authentication document Exists",
+      data: { device_info: userDeviceInfoData.data.device_info },
```
[the remark this answers is not in the document]
Comment (shreya-mishra) [reply controllers/auth.js]: resolved it in last commit

File: middlewares/validators/qrCodeAuth.js [hunk 3]
```diff
@@ -46,8 +46,23 @@ const validateFetchingUserDocument = async (req, res, next) => {
...
+const validateFetchingUserDeviceStatus = async (req, res, next) => {
+  const schema = joi.object().strict().keys({
+    user_id: joi.string().required(),
+  });
+
+  try {
+    await schema.validateAsync(req.query);
+    next();
+  } catch (error) {
+    logger.error(`Invalid Query Parameters Passed`);
```
[the remark this answers is not in the document]
Comment (shreya-mishra) [reply middlewares/validators/qrCodeAuth.js]: I agree , resolved it in last commit

File: middlewares/validators/qrCodeAuth.js [hunk 4]
```diff
@@ -46,8 +46,23 @@ const validateFetchingUserDocument = async (req, res, next) => {
...
+    res.boom.badRequest(`Invalid Query Parameters Passed`);
```
[the remark this answers is not in the document]
Comment (shreya-mishra) [reply middlewares/validators/qrCodeAuth.js]: +1

**Comment by shreya-mishra** [comment]
Raised a different PR https://github.com/Real-Dev-Squad/website-backend/pull/1400

**Pull request closed without merging**